shell/common/shell_unittests.cc

```diff
@@ -1014,5 +1015,114 @@ TEST_F(ShellTest, Screenshot) {
   DestroyShell(std::move(shell));
 }
 
+static bool MemsetPatternSetOrCheck(uint8_t* buffer,
+                                    size_t size,
+                                    bool set_or_check) {
+  if (buffer == nullptr) {
+    return false;
+  }
+
+  auto pattern4 = reinterpret_cast<const uint8_t*>("dErP");
+  uint8_t* start = buffer;
+  uint8_t* p = buffer;
+
+  while ((start + size) - p >= 4) {
+    if (set_or_check) {
+      memmove(p, pattern4, 4);
+    } else {
+      if (memcmp(pattern4, p, 4) != 0) {
+        return false;
+      }
+    }
+    p += 4;
+  }
+
+  if ((start + size) - p != 0) {
+    if (set_or_check) {
+      memmove(p, pattern4, (start + size) - p);
+    } else {
+      if (memcmp(pattern4, p, (start + size) - p) != 0) {
+        return false;
+      }
+    }
+  }
+
+  return true;
+}
+
+TEST_F(ShellTest, CanConvertToAndFromMappings) {
+  const size_t buffer_size = 2 << 20;
+
+  uint8_t* buffer = static_cast<uint8_t*>(::malloc(buffer_size));
+  ASSERT_NE(buffer, nullptr);
+  ASSERT_TRUE(MemsetPatternSetOrCheck(buffer, buffer_size, true /* set */));
```

Comment:
Wonder if you should just chuck in an two-valued enum instead?

Reply:
Done.